packages/inventory/src/Stocktake/api/api.ts

```diff
@@ -111,11 +108,20 @@ export const getStocktakeQueries = (sdk: Sdk, storeId: string) => ({
   updateLines: async (draftStocktakeLines: DraftStocktakeLine[]) => {
     const input = {
       storeId,
+      deleteStocktakeLines: draftStocktakeLines
+        .filter(
+          ({ countThisLine, isUpdated, isCreated }) =>
+            !isCreated && isUpdated && !countThisLine
+        )
+        .map(stocktakeParser.line.toDelete),
       insertStocktakeLines: draftStocktakeLines
-        .filter(({ isCreated }) => isCreated)
+        .filter(({ isCreated, countThisLine }) => isCreated && countThisLine)
         .map(stocktakeParser.line.toInsert),
       updateStocktakeLines: draftStocktakeLines
-        .filter(({ isCreated, isUpdated }) => !isCreated && isUpdated)
+        .filter(
+          ({ countThisLine, isCreated, isUpdated }) =>
+            !isCreated && isUpdated && countThisLine
+        )
```

Comment:
Aaaand the actual change the issue was for 🙄 

Only insert/update if `countThisLine` is true. Delete if `isUpdated` and `countThisLine` is true